docs/platform-external/use-case-oci-upi.md

```diff
@@ -0,0 +1,1195 @@
+rateLimiter:
+  rateLimitQPSRead: 20.0
+  rateLimitBucketRead: 5
+  rateLimitQPSWrite: 20.0
+  rateLimitBucketWrite: 5
+EOF
+
+# Create the secret manifest
+cat <<EOF1 > ${INSTALL_DIR}/manifests/oci-01-ccm-00-secret.yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: oci-cloud-controller-manager
+  namespace: $OCI_CCM_NAMESPACE
+data:
+  cloud-provider.yaml: $(base64 -w0 < ./cloud-provider.yaml)
+EOF1
+```
+
+!!! danger "Bug"
+    The CCM is raising the error below in the startup/logs. The secret data `config.yaml` has been duplicated to prevent this error.
+
+    ```
+    ERROR   oci/ccm.go:122  Metrics collection could not be enabled {"component": "cloud-controller-manager", "error": "failed to load configuration file at path /etc/oci/config.yaml: open /etc/oci/config.yaml: no such file or directory"
+    ```
+
+!!! warning "Question"
+    - Is it possible to use NSG instead of SecList in Load Balancer?
+
+- Download manifests from [OCI CCM's Github](https://github.com/oracle/oci-cloud-controller-manager) and save it in the directory `${INSTALL_DIR}/manifests`:
+
+```bash
+export CCM_RELEASE=v1.25.0
+
+wget https://github.com/oracle/oci-cloud-controller-manager/releases/download/${CCM_RELEASE}/oci-cloud-controller-manager-rbac.yaml -O oci-cloud-controller-manager-rbac.yaml
+
+wget  https://github.com/oracle/oci-cloud-controller-manager/releases/download/${CCM_RELEASE}/oci-cloud-controller-manager.yaml -O oci-cloud-controller-manager.yaml
+```
+
+- Patch the RBAC file setting the correct namespace in the `ServiceAccount`:
+
+```bash
+./yq ". | select(.kind==\"ServiceAccount\").metadata.namespace=\"$OCI_CCM_NAMESPACE\"" oci-cloud-controller-manager-rbac.yaml > ./oci-cloud-controller-manager-rbac_patched.yaml
+```
+
+- Patch the RBAC file setting the correct namespace in the `ServiceAccount`:
+
+```bash
+cat << EOF > ./oci-ccm-rbac_patch_crb-subject.yaml
+- kind: ServiceAccount
+  name: cloud-controller-manager
+  namespace: $OCI_CCM_NAMESPACE
+EOF
+
+./yq eval-all -i ". | select(.kind==\"ClusterRoleBinding\").subjects *= load(\"oci-ccm-rbac_patch_crb-subject.yaml\")" ./oci-cloud-controller-manager-rbac_patched.yaml
+```
+
+- Split the RBAC manifest file
+
+>> debug/investigation the manifest render errors on bootstrap stage
+
+```bash
+./yq -s '"./oci-01-ccm-01-rbac_" + $index' ./oci-cloud-controller-manager-rbac_patched.yaml &&\
+mv -v ./oci-01-ccm-01-rbac_*.yml ${INSTALL_DIR}/manifests/
+```
+
+- Patch the CCM DaemonSet manifest setting the namespace, append the tolerations, mount CA, and add env vars for the kube API URL used in OpenShift:
+
+```bash
+# Create the pod template patch
+cat <<EOF > ./oci-cloud-controller-manager-ds_patch1.yaml
+metadata:
+  namespace: $OCI_CCM_NAMESPACE
+spec:
+  template:
+    spec:
+      tolerations:
+        - key: node.kubernetes.io/not-ready
+          operator: Exists
+          effect: NoSchedule
+EOF
+
+# Create the containers' patch (TODO merge both patches)
```

Comment:
we may need to add some flags to the CCM binary because we deploy in another namespace, see: https://github.com/openshift/assisted-test-infra/pull/2200#discussion_r1248682632